lib/puppet/graph/simple_graph.rb

```diff
@@ -481,70 +483,49 @@ class << self
   end
   self.use_new_yaml_format = false
 
-  # Stub class to allow graphs to be represented in YAML using the old
-  # (version 2.6) format.
-  class VertexWrapper
-    attr_reader :vertex, :adjacencies
-    def initialize(vertex, adjacencies)
-      @vertex = vertex
-      @adjacencies = adjacencies
-    end
-
-    def inspect
-      { :@adjacencies => @adjacencies, :@vertex => @vertex.to_s }.inspect
+  def initialize_from_hash(hash)
+    initialize
```

Comment:
I tried without it. Doesn't work. I suspect that YAML just allocates, then calls initialize_from_hash.